### PR TITLE
feat: remirror V3

### PR DESCRIPTION
**Pull request by ocavue**
### Checklist

- [ ] I have read the [**contributing**](https://github.com/remirror/remirror/blob/HEAD/docs/contributing.md) document.
- [ ] My code follows the code style of this project and `pnpm fix` completed successfully.
- [ ] I have updated the documentation where necessary.
- [ ] New code is unit tested and all current tests pass when running `pnpm test`.

### CODE REVIEW

**Comment by ocavue** [comment]
@whawker FYI, you can already install `remirror@3.0.0-beta.0`

File: packages/remirror__core/src/extension/extension-decorator.ts [hunk 1]
```diff
@@ -154,7 +154,14 @@ export type ExtensionDecoratorOptions<Options extends Shape = EmptyShape> = Defa
...
   options: ExtensionDecoratorOptions<Options>,
 ) {
-  return <Type extends AnyExtensionConstructor>(ReadonlyConstructor: Type): Type => {
+  return <Type extends AnyExtensionConstructor>(
+    ReadonlyConstructor: Type,
+    context: ClassDecoratorContext<Type>,
+  ): Type => {
+    if (context.kind !== 'class') {
+      throw new Error(`The extension decorator can only be used on a class.`);
+    }
```
Comment (whawker) [inline packages/remirror__core/src/extension/extension-decorator.ts:163]: [In the docs](https://remirror.io/docs/concepts/extension#extension-1) we mention that you can use decorators as a normal function

Screenshot 20
23-11-02 at 18 13 35

For this to be true now, we would need to use (at a minimum)

```diff
- extension({ defaultSettings: { color: 'red' } })(ExampleExtension);
+ extension({ defaultSettings: { color: 'red' } })(ExampleExtension, { kind: 'class' });
```

Should we let `context` default to `{ kind: 'class' }` if undefined? Or should we update the docs instead?

Comment (ocavue) [reply packages/remirror__core/src/extension/extension-decorator.ts:163]: we let context default to `{ kind: 'class' }`

Comment (whawker) [reply packages/remirror__core/src/extension/extension-decorator.ts:163]: Should an appropriate default be added for all decorators, or just this `extension` one?

Comment (ocavue) [reply packages/remirror__core/src/extension/extension-decorator.ts:163]: I think all decorators from Remirror should have default `kind`

**Pull request merged**